diff --git a/lib/worker.js b/lib/worker.js
index 03d02d18bb90d434372cfea2fe772ae520101390..296244b5c28800dc363ec6e0e5db04cb306c3373 100644
--- a/lib/worker.js
+++ b/lib/worker.js
@@ -44,7 +44,7 @@ async function start () {
     if ((error.code === 'ENOTDIR' || error.code === 'ERR_MODULE_NOT_FOUND') &&
      filename.startsWith('file://')) {
       worker = realRequire(decodeURIComponent(filename.replace('file://', '')))
-    } else if (error.code === undefined) {
+    } else if (error.code === undefined || error.code === 'ERR_VM_DYNAMIC_IMPORT_CALLBACK_MISSING') {
       // When bundled with pkg, an undefined error is thrown when called with realImport
       worker = realRequire(decodeURIComponent(filename.replace(process.platform === 'win32' ? 'file:///' : 'file://', '')))
     } else {
